index.php

```diff
@@ -66,14 +70,22 @@
             <input type="text" name="price" placeholder="price (pence)" required>
             <input type="text" name="remaining" placeholder="remaining (%)" required>
             <input type="submit" value="Submit">
+            <p><?php
+                if($item && $category && $price && $remaining) {
```

Comment:
Looks like this fails silently instead of providing an error message, is that coming later?

Reply:
it does throw type error, when wrong data type is passed. When input is empty and submitted, input itself will ask you to fill it in